libraries/c_sdk/aws/shadow/test/unit/aws_iot_tests_shadow_api.c

```diff
@@ -301,10 +354,43 @@ static size_t _sendSuccess( void * pSendContext,
         else
         {
             mqttPacket.u.pIncomingPublish = &deserializedPublish;
-            mqttPacket.pRemainingData = ( uint8_t * ) pMessage + ( messageLength - mqttPacket.remainingLength );
+
+            /* Check if MQTT PUBLISH packet is fully received. */
+            if( _publishContext.dataLength == 0 )
+            {
+                mqttPacket.pRemainingData = ( uint8_t * ) pMessage + ( messageLength - mqttPacket.remainingLength );
+            }
+            else
+            {
+                AwsIotShadow_Assert( _publishContext.pData != NULL );
+                /* Copy the remaining part of the MQTT PUBLISH. */
+                memcpy( ( void * ) ( _publishContext.pData + _publishContext.dataLength ), pMessage, messageLength );
+                _publishContext.dataLength += messageLength;
+
+                /* Read the packet type, which is the first byte in the message. */
+                mqttPacket.type = *_publishContext.pData;
```

Comment:
_publishContext.pData[ 0 ] seems more readable for reading first byte.

Reply:
Yea. Will address it. Thanks